src/main/java/com/waytta/saltAPI.java

```diff
@@ -0,0 +1,95 @@
+package com.waytta;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import jenkins.security.MasterToSlaveCallable;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+
+
+class saltAPI extends MasterToSlaveCallable<JSONObject, IOException> {
+    private static final long serialVersionUID = 1L;
+    private String targetURL;
+    private JSONObject urlParams;
+    private String auth;
+
+    public saltAPI(String targetURL, JSONObject urlParams, String auth) {
+        this.targetURL = targetURL;
+        this.urlParams = urlParams;
+        this.auth = auth;
+    }
+    
+    public JSONObject call() throws IOException {
+        HttpURLConnection connection = null;
+        JSONObject responseJSON = new JSONObject();
+
+        try {
+            // Create connection
+            URL url = new URL(targetURL);
+            connection = (HttpURLConnection) url.openConnection();
```

Comment:
Possible Class Cast exception if a user specifies another protocol (e.g. "file:///" or "jar://" even if it id weird)